src/goto-instrument/nondet_static.cpp

```diff
@@ -38,23 +38,25 @@ void nondet_static(
       const symbol_exprt &sym=to_symbol_expr(
           to_code_assign(instruction.code).lhs());
 
-      // is it a __CPROVER_* variable?
-      if(has_prefix(id2string(sym.get_identifier()), CPROVER_PREFIX))
-        continue;
-
       // static lifetime?
       if(!ns.lookup(sym.get_identifier()).is_static_lifetime)
         continue;
 
+      // should not be nondet-initialized, e.g., proprietary variables?
+      if(
+        ns.lookup(sym.get_identifier())
+          .type.get_bool(ID_C_no_nondet_initialization_allowed))
```

Comment:
Presumably also check for ID_C_constant?

Reply:
Constants are checked below, using `is_constant_or_has_constant_components`